internal/backend/crypto/gpg/cli/encrypt.go

```diff
@@ -25,25 +28,37 @@ func (g *GPG) Encrypt(ctx context.Context, plaintext []byte, recipients []string
 		args = append(args, "--trust-model=always")
 	}
 
+	if len(recipients) == 0 {
+		debug.Log("recipients list is empty!")
+	} else {
+		debug.Log("recipients are %s", recipients)
+	}
+	var badRecipients []string
 	for _, r := range recipients {
 		kl, err := g.listKeys(ctx, "public", r)
 		if err != nil {
 			debug.Log("Failed to check key %s. Adding anyway. %s", err)
 		} else if len(kl.UseableKeys(gpg.IsAlwaysTrust(ctx))) < 1 {
-			out.Printf(ctx, "Not using invalid key %s for encryption. (Check its expiration date or its encryption capabilities.)", r)
+			badRecipients = append(badRecipients, r)
+			errmsg := fmt.Sprintf("Not using invalid key %s for encryption. Check its expiration date, its encryption capabilities and trust.", r)
+			debug.Log(errmsg)
+			out.Printf(ctx, errmsg)
 
 			continue
 		}
+		debug.Log("adding recipient %s", r)
 		args = append(args, "--recipient", r)
 	}
-
 	buf := &bytes.Buffer{}
+	if len(badRecipients) == len(recipients) && len(recipients) != 0 {
+		return buf.Bytes(), errors.New("no valid and trusted recipients were found!")
+	}
 
 	cmd := exec.CommandContext(ctx, g.binary, args...)
 	cmd.Stdin = bytes.NewReader(plaintext)
-	// the encrypted blob is written to stdout
-	cmd.Stdout = buf
-	cmd.Stderr = os.Stderr
+	// the encrypted blob and errors are printed to the log file, and to stdout
+	cmd.Stdout = io.MultiWriter(buf, debug.LogWriter)
```

Comment:
I wonder if we really want to write the encrypted blobs to the log file, this seems like it may contain sensitive data and information needed to debug gpg output should probably already be in `Stderr`, no?

Reply:
I think writing encrypted blobs to the log is ok, but I wonder if it's useful?
Can you easily extract the encrypted blob from the log and operate on it with gpg?

Reply:
> I think writing encrypted blobs to the log is ok, but I wonder if it's useful?

As I said in my previous comment: In the experience I had with `gopass-jsonapi listen`, I had available only the log opened in a file, and the stdout and stderr of the program were hidden in by the browser. Hence I would argue that it is useful to print as much information to the log.

> Can you easily extract the encrypted blob from the log and operate on it with gpg?

If we are talking about the user's gpg which has available the user's private keys etc, then yes. Other people shouldn't be able to decrypt these blobs if the user has their private keys kept private.